Sharpmake/Project.Configuration.cs

```diff
@@ -2775,7 +2786,10 @@ internal void Resolve(Resolver resolver)
                 {
                     customFileBuildStep.Resolve(resolver);
                     Util.ResolvePath(Project.SourceRootPath, ref customFileBuildStep.KeyInput);
-                    Util.ResolvePath(Project.SourceRootPath, ref customFileBuildStep.Executable);
+                    if(customFileBuildStep.ResolveExecutable)
```

Comment:
Hi, thanks for getting back to me!
Maybe I have configured something wrong but when that resolve happens, the project path gets appended to the executable name even if there are no `[project.XXXX]` variables to resolve.

![image](https://github.com/ubisoft/Sharpmake/assets/3481015/89db174e-af05-43b5-9d27-c65658879e66)

Reply:
Hello,
Sharpmake assumes that the executable is in the SourceRootPath folder since you didn't specified an absolute path. In general however, from a build system point of view, relying on PATH to find your compiler is not really a recommended approach. 

There is several reasons to this:
- Build will fail if your compiler is missing from path
- You will have build reproducibility issues(two machine could have different version of the compiler installed)

In general at ubisoft for these kind of things we generally use nugets for our various compilers and sdks(vs2022, console sdks etc). In some simple case with only a single .exe we sometimes publish the .exe in our source control. By doing these we insure build reproducibility. You should never rely on what is installed on a pc and especially not rely on what is in the PATH variable.

Reply:
Thank you for the explanation. Those are all valid points when it comes to real production, in my case its my personal codebase and when I use it anywhere I know about the limitations. This is mainly because I don't want to check in glslang because I don't want to waste space on my limited hosted server and git is really bad with binaries.

I'm sure you could find use cases for it across a team in specific situations but I'd imagine those are rare. If you think this sort of goes against the design of sharpmake then we can close the PR however if you think the base idea is valid but needs some changes I'd appreciate any feedback :) 

Thank you for taking the time reviewing this.

Reply:
What I suggest you to do is if you know the full path to glslang, specify it in sharpmake. I think this will fix your problem

Reply:
Oh in terms of workarounds or fixing the problem, it also works to just have a custom class for customfildbuildstep. However both cases add a dependency to AdditionalInputs field hence why I have the other flag in there too.